### PR TITLE
gnutls: enable latest version on Darwin

### LINKED ISSUE #58481: insecure GnuTLS on darwin
Darwin has been using old GnuTLS version due to [some issues](https://github.com/NixOS/nixpkgs/commit/d6454e6a1), but that means it's vulnerable...

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/commit/d6454e6a1593f12e17d982a1ad30c97ca7cf130a was held back because https://github.com/NixOS/nixpkgs/issues/24693, which was implemented in https://github.com/NixOS/nixpkgs/pull/27598 cc: @vcunat

Fixes https://github.com/NixOS/nixpkgs/issues/58481

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by matthewbauer — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10273,13 +10273,9 @@ in
               then callPackage ../development/libraries/gnu-efi { }
             else null;
 
-  gnutls = callPackage
-    (if stdenv.isDarwin
-      # Avoid > 3.5.10 due to frameworks for now; see discussion on:
-      # https://github.com/NixOS/nixpkgs/commit/d6454e6a1
-      then ../development/libraries/gnutls/3.5.10.nix
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix]: You will need to remove this file now as it is not referenced by anything else.

Comment (marsam) [reply pkgs/top-level/all-packages.nix]: good catch. Fixed, thanks!

**Comment by vcunat** [comment]
@GrahamcOfBorg build gnutls

**Review by fpletz — APPROVED** [review]

**Comment by vcunat** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/58611

**Pull request closed without merging**